mqtt/mqtt-bridge/src/messages.rs

```diff
@@ -84,15 +84,13 @@ impl<S> StoreMqttEventHandler<S> {
         if let Some(mapper) = self.topic_mappers_updates.get(sub) {
             self.topic_mappers.insert(sub.to_owned(), mapper);
         } else {
-            debug!("subscription ack for {} not expected", sub);
+            warn!("unexpected subscription ack for {}", sub);
         };
     }
 
     fn update_unsubscribed(&mut self, sub: &str) {
-        if self.topic_mappers_updates.contains_key(sub) {
-            self.topic_mappers.remove(sub);
-        } else {
-            debug!("unsubscription ack for {} not expected", sub);
+        if self.topic_mappers_updates.remove(sub).is_none() {
```

Comment:
it should be removed from topic_mappers, not from topic_mappers_updates

Reply:
that shows that there is no test for unsubscribe, can you please add or I ca add one

Reply:
Yeah, this is a mistake here